google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingHandler.java

```diff
@@ -428,13 +455,33 @@ public Boolean getAutoPopulateMetadata() {
    * Enable/disable redirection to STDOUT. If set to {@code true}, logs will be printed to STDOUT in
    * the Json format that can be parsed by the logging agent. If set to {@code false}, logs will be
    * ingested to Cloud Logging by calling Logging API.
+   *
+   * @deprecated Use setLogTarget.
    */
+  @Deprecated
   public void setRedirectToStdout(boolean value) {
-    this.redirectToStdout = value;
+    this.logTarget = value ? LogTarget.STDOUT : LogTarget.CLOUD_LOGGING;
   }
 
+  /*
+   * @deprecated Use getLogTarget.
+   */
+  @Deprecated
   public Boolean getRedirectToStdout() {
-    return redirectToStdout;
+    return this.logTarget == LogTarget.STDOUT;
+  }
+
+  /**
+   * Configure the destination for ingested logs. If set to STDOUT or STDERR, logs will be printed
+   * to the corresponding stream in the Json format that can be parsed by the logging agent. If set
+   * to CLOUD_LOGGING, logs will be sent directly to the Google Cloud Logging API.
+   */
+  public void setLogTarget(LogTarget value) {
```

Comment:
I think generally sending logs to *both* stdout and stderr would be unusual. I think the most common consensus e.g., [on StackOverflow](https://stackoverflow.com/questions/4919093/should-i-log-messages-to-stderr-or-stdout) is that logs should *only* go to `stderr`; `stdout` is for the program's primary output, if it has it. I've also [seen](https://github.com/bloomberg/bde/blob/master/groups/bal/ball/ball_fileobserver.h) stderr or stdout + file with severity-based thresholds for each. But doing both stderr and stdout seems unusual.

In my opinion to avoid unnecessary complexity in the interface, it's best to tell anyone who wants some permutation of (stdout, stderr, file, Cloud Logging) based on severity to *instantiate two handlers*. You can already express this in JUL; it has built-in systems to manage the log multiplexing. I think building multiplexing into the handler is a mistake that, e.g., leads one to later simplify the [two-output handler](https://github.com/bloomberg/bde/blob/master/groups/bal/ball/ball_fileobserver.h) down to [a simple one-output one](https://github.com/bloomberg/bde/blob/master/groups/bal/ball/ball_fileobserver2.h).

Reply:
... Regarding adding `setRedirectToStderr`: This is somewhat hinges on decision of whether we want to build multiplexing in the `LoggingHandler`. IMO we shouldn't, a LoggingHandler instance should have one output at a time.

If we do that, the output control is a single variable: an enum, not a boolean. Because of that in this PR I **deprecated** `setRedirectToStdout` since it takes a boolean, not the new `LogTarget` enum. Adding `setRedirectToStderr` would add another already-deprecated method.

Of course we could continue to *not do multiplexing in `LoggingHandler`*, while also leaving ``setRedirectToStdout` **un**deprecated, and add `setRedirectToStdout`... they'd just be semantically misleading since they'd be acting like setters for two booleans when they're actually setters on the same enum.

Reply:
> ... Regarding adding `setRedirectToStderr`: This is somewhat hinges on decision of whether we want to build multiplexing in the `LoggingHandler`. IMO we shouldn't, a LoggingHandler instance should have one output at a time.
> 
> If we do that, the output control is a single variable: an enum, not a boolean. Because of that in this PR I **deprecated** `setRedirectToStdout` since it takes a boolean, not the new `LogTarget` enum. Adding `setRedirectToStderr` would add another already-deprecated method.
> 
> Of course we could continue to _not do multiplexing in `LoggingHandler`_, while also leaving `setRedirectToStdout` **un**deprecated, and add `setRedirectToStdout`... they'd just be semantically misleading since they'd be acting like setters for two booleans when they're actually setters on the same enum.

Yep, this is indeed a bit unusual... I must admit that `setLogTarget()` is confusing as well - library user might not understand immediately that `setLogTarget()` and `setRedirectToStdout()` are mutually exclusive...

Reply:
Okay, added an explicit comment to both `setLogTarget` and `setRedirectToStdout` that they're mutually exclusive.

I'd already marked `setRedirectToStdout` as deprecated.